### DISCUSSION TITLE
PWM Parameters

### DISCUSSION THREAD

**Discussion by jimcraig47** [micropython/micropython#10759] [RP2040 / Pico]
from machine import Pin, PWM
freq = 1000
led = PWM(Pin(10), freq)

When I run this code on an ESP32 it runs fine. Trying to understand why when I run the same code on a Raspberry Pi Pico I get the "TypeError: function takes 1 positional arguments but 2 were given" for the last line.

**Comment by robert-hh** [comment]
That is indeed an inconsistency. At the RP2 port the PWM constructor does not accept additional arguments. You have to do it step-by-step, e.g.:
```
from machine import Pin, PWM
freq = 1000
led = PWM(Pin(10))
led.freq(freq)
led.duty_u16(32768)
```

**Reply by jimcraig47** [reply]
Thanks

**Comment by ned-pcs** [comment]
It currently allows one positional argument (`freq`), and several keyword arguments (`duty_u16`, `duty_ns`, and `invert`):
```python
pwm1 = machine.PWM(machine.Pin(28), freq=1000, invert=False)
```
